lib/routes/routePOST.js

```diff
@@ -4,16 +4,32 @@ import api from '../api/api';
 import routesUtils from './routesUtils';
 import pushMetrics from '../utilities/pushMetrics';
 
+const MAX_POST_LENGTH = 1024 * 1024 * 0.5; // 512 KB
+
 /* eslint-disable no-param-reassign */
 export default function routePOST(request, response, log, utapi) {
     log.debug('routing request', { method: 'routePOST' });
-    request.post = '';
 
+    const post = [];
+    let postLength = 0;
     request.on('data', chunk => {
-        request.post += chunk.toString();
+        post.push(chunk);
+        postLength += chunk.length;
+        // Sanity check on post length
+        if (postLength > MAX_POST_LENGTH) {
+            log.warn('post length is too long',
+                         { postLength });
+            request.pause();
```

Comment:
because the client will not read her socket for the response until she has terminated writing the whole body.
If you want the client to figure out what is going on, you should:
* mandate the use of expect 100 continue feature, spot the bad request earlier and abort the body transfer prior to its initialization
* or read the whole body, even if there is a 4XX or 5XX, and then send the response.
if you cannot do any of these, you must shutdown the tcp/ip connection, but it's not a good idea.